### PR TITLE
HPCC-21719 Reorganize dafilesrv and client code

### PR DESCRIPTION
**Pull request by jakesmith**
Prompted by circular dependency issues (effecting window compiles)
but have wanted to do anyway.
Separates client code from dafilesrv server code.
Streaming code in own module/library.
Remote code unrelated to dafilesrv separated (left in libremote)

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: dali/dfuplus/dfuplus.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 #ifdef DAFILESRV_LOCAL
 #include "rmtfile.hpp"
+#include "rmtclient.hpp"
 #include "sockfile.hpp"
```
Comment (richardkchapman) [inline dali/dfuplus/dfuplus.cpp]: I thought sockfile.hpp was deleted?

**Comment by richardkchapman** [comment]
There seems to be a new file dali/ft/.filecopy.cpp.swp that probably should not be included

File: dali/dfuplus/dfuplus.cpp (hunk 1 above)
Comment (jakesmith) [reply dali/dfuplus/dfuplus.cpp]: no, it remains the main dafilesrv server component and in lib dafsserver.
I considered renaming it, feeling I prob. still should

**Comment by jakesmith** [comment]
>There seems to be a new file dali/ft/.filecopy.cpp.swp that probably should not be included

Now removed and squashed in the changed.

File: fs/dafsclient/dafscommon.hpp [hunk 2]
```diff
@@ -18,6 +18,14 @@
 #ifndef DAFSCOMMON_HPP
 #define DAFSCOMMON_HPP
 
+#define FILESRV_VERSION 25 // don't forget VERSTRING in sockfile.cpp
+// JCSMORE very temporary! (should prob used FILESRV_VERSION)
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply fs/dafsclient/dafscommon.hpp]: er no, I thought I'd removed it already.
Checking..

[the remark this answers is not in the document]
Comment (jakesmith) [reply fs/dafsclient/dafscommon.hpp]: I had some unpushed changes that were in a separate git repo.
I've now pushed them as a separate commit.

**Review by ghalliday — COMMENTED** [review]
@jakesmith a few unnecessary dependencies.  Otherwise looks good.

File: esp/clients/wsdfuaccess/CMakeLists.txt [hunk 3]
```diff
@@ -65,9 +67,13 @@ target_link_libraries ( wsdfuaccess
          jlib
          xmllib 
          esphttp
-         remote
+         dafsclient
+         dafsserver
+         dafsstream
          thorhelper
          dalibase
+         dafsclient
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply esp/clients/wsdfuaccess/CMakeLists.txt]: agreed, will remove.

File: esp/clients/wsdfuaccess/CMakeLists.txt [hunk 4]
```diff
@@ -65,9 +67,13 @@ target_link_libraries ( wsdfuaccess
          jlib
          xmllib 
          esphttp
-         remote
+         dafsclient
+         dafsserver
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply esp/clients/wsdfuaccess/CMakeLists.txt]: It's needed by the unittests (which creates a server thread), could make conditional on CPPUNIT..

File: testing/unittests/CMakeLists.txt [hunk 5]
```diff
@@ -45,7 +45,8 @@ include_directories (
          ./../../system/jlib
          ./../../system/mp
          ./../../system/libbase58
-         ./../../common/remote
+         ./../../fs/dafsclient
+         ./../../fs/dafsclient
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply testing/unittests/CMakeLists.txt]: will remove.

File: esp/clients/wsdfuaccess/CMakeLists.txt [hunk 6]
```diff
@@ -39,7 +39,9 @@ include_directories (
          ${HPCC_SOURCE_DIR}/system/security/shared
          ${HPCC_SOURCE_DIR}/system/security/securesocket
          ${HPCC_SOURCE_DIR}/system/security/cryptohelper
-         ${HPCC_SOURCE_DIR}/common/remote
+         ${HPCC_SOURCE_DIR}/fs/dafsclient
+         ${HPCC_SOURCE_DIR}/fs/dafsstream
+         ${HPCC_SOURCE_DIR}/fs/dafsserver
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply esp/clients/wsdfuaccess/CMakeLists.txt]: it's needed by unittests

File: esp/clients/wsdfuaccess/CMakeLists.txt [hunk 7]
```diff
@@ -65,9 +67,13 @@ target_link_libraries ( wsdfuaccess
...
          xmllib 
          esphttp
-         remote
+         dafsclient
+         dafsserver
+         dafsstream
          thorhelper
          dalibase
+         dafsclient
+         environment
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply esp/clients/wsdfuaccess/CMakeLists.txt]: It should be required for getAccessibleServiceURLList() afaics.

**Comment by jakesmith** [comment]
@ghalliday - pushed review changes.

**Comment by richardkchapman** [comment]
@AttilaVamos Do you know what happened to smoketest?

**Comment by jakesmith** [comment]
Attila is doing some tests to make smoketest faster, and hit issues to do with that.
It doesn't have to be tested on this PR though.

**Comment by AttilaVamos** [comment]
Now I restored the Smoketest and this PR will be re-tested.

**Comment by jakesmith** [comment]
@ghalliday - squashed as requested.

**Pull request merged**